### PR TITLE
Store targets and observations in DB and sync clients with DB

### PR DESCRIPTION
**Pull request by rpiaggio**
Also, renamed conditions->constraints.

Animated GIF screen capture is a bit large to upload to GitHub, but here it is in MP4: https://drive.google.com/file/d/1F_phHQYh6bb1qoWsPTHd7LsCVn-4-K2j/view?usp=sharing

### CODE REVIEW

**Comment by cquiroz** [comment]
The capture looks amazing

File: common/src/main/scala/explore/model/decoders.scala [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+import gsp.math.Declination
+import gsp.math.Coordinates
+import gsp.math.Epoch
+import gsp.math.ProperMotion
+import explore.model.enum.ObsStatus
+import java.time.Duration
+
+object decoders {
+
+  implicit def enumDecoder[E: Enumerated]: Decoder[E] =
```
Comment (cquiroz) [inline common/src/main/scala/explore/model/decoders.scala:25]: This is great. We should consider whether these should go to gsp-core eventually

File: db.sql [hunk 2]
```diff
@@ -0,0 +1,88 @@
+CREATE EXTENSION IF NOT EXISTS "uuid-ossp";
```
Comment (cquiroz) [inline db.sql:1]: Is this here just to keep a record? or do we use it to build the db?

File: db.sql [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+  'Proposed',
+  'Approved',
+  'ForReview',
+  'Ready',
+  'Ongoing',
+  'Observed'
+);
+
+CREATE TABLE constraints (
+  id UUID PRIMARY KEY DEFAULT uuid_generate_v4(),
```
Comment (cquiroz) [inline db.sql:48]: We don't need to change this but there is a requirement for "memorable" urls. we may want to use a human readable id eventually/aditionally

File: observationtree/src/main/scala/explore/observationtree/TargetObsList.scala [hunk 4]
```diff
@@ -32,12 +32,12 @@ import explore.model.Focused
...
+import TargetObsQueries._
 
 final case class TargetObsList(
-  targets:        List[SiderealTarget],
-  observations:   View[List[ExploreObservation]],
-  focused:        ViewOpt[Either[SiderealTarget.Id, ExploreObservation.Id]],
-  onTargetSelect: SiderealTarget.Id => Callback
+  targets:      List[SiderealTarget],
+  observations: View[List[ExploreObservation]],
+  focused:      View[Option[Either[SiderealTarget.Id, ExploreObservation.Id]]]
```
Comment (cquiroz) [inline observationtree/src/main/scala/explore/observationtree/TargetObsList.scala:40]: Why the Either? Is this to indicate we either clicked on a given target or one of the obs attached to the target?

Comment (rpiaggio) [reply observationtree/src/main/scala/explore/observationtree/TargetObsList.scala:40]: Exactly. And the `Option` is for the case where we are in this page but no target or obs has been clicked yet.

File: db.sql (hunk 3 above)
Comment (rpiaggio) [reply db.sql:48]: Ok. We can change to anything as long as it's unique.

File: db.sql (hunk 2 above)
Comment (rpiaggio) [reply db.sql:1]: It provides the `uuid_generate_v4` function.

File: common/src/main/scala/explore/model/decoders.scala (hunk 1 above)
Comment (rpiaggio) [reply common/src/main/scala/explore/model/decoders.scala:25]: Yes, good idea. We will probably use this from other apps too.

**Pull request merged**